gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Acl.java

```diff
@@ -0,0 +1,400 @@
+    Access toPb() {
+      return new Access().setDomain(domain);
+    }
+  }
+
+  /**
+   * Class for a BigQuery Group entity.
+   */
+  public static final class Group extends Entity {
+
+    private static final String PROJECT_OWNERS = "projectOwners";
+    private static final String PROJECT_READERS = "projectReaders";
+    private static final String PROJECT_WRITERS = "projectWriters";
+    private static final String ALL_AUTHENTICATED_USERS = "allAuthenticatedUsers";
+    private static final long serialVersionUID = 5146829352398103029L;
+    
+    private final String identifier;
+
+    /**
+     * Creates a Group entity given its identifier. Identifier can be either a special group
+     * identifier ({@code projectOwners}, {@code projectReaders}, {@code projectWriters} and
+     * {@code allAuthenticatedUsers}) or a group email.
+     */
+    public Group(String identifier) {
+      super(Type.GROUP);
+      this.identifier = identifier;
+    }
+
+    /**
+     * Returns group's identifier, can be either a special group identifier ({@code projectOwners},
+     * {@code projectReaders}, {@code projectWriters} and {@code allAuthenticatedUsers}) or a group
+     * email.
+     */
+    public String identifier() {
+      return identifier;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      Group group = (Group) o;
+      return Objects.equals(type(), group.type()) && Objects.equals(identifier, group.identifier);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(type(), identifier);
+    }
+
+    @Override
+    public String toString() {
+      return toPb().toString();
+    }
+
+    @Override
+    Access toPb() {
+      switch (identifier) {
+        case PROJECT_OWNERS:
+          return new Access().setSpecialGroup(PROJECT_OWNERS);
+        case PROJECT_READERS:
+          return new Access().setSpecialGroup(PROJECT_READERS);
+        case PROJECT_WRITERS:
+          return new Access().setSpecialGroup(PROJECT_WRITERS);
+        case ALL_AUTHENTICATED_USERS:
+          return new Access().setSpecialGroup(ALL_AUTHENTICATED_USERS);
+        default:
+          break;
+      }
+      return new Access().setGroupByEmail(identifier);
+    }
+
+    /**
+     * Returns a Group entity representing all project's owners.
+     */
+    public static Group ofProjectOwners() {
+      return new Group(PROJECT_OWNERS);
+    }
+
+    /**
+     * Returns a Group entity representing all project's readers.
+     */
+    public static Group ofProjectReaders() {
+      return new Group(PROJECT_READERS);
+    }
+
+    /**
+     * Returns a Group entity representing all project's writers.
+     */
+    public static Group ofProjectWriters() {
+      return new Group(PROJECT_WRITERS);
+    }
+
+    /**
+     * Returns a Group entity representing all project's users.
```

Comment:
According to [this](https://cloud.google.com/bigquery/docs/reference/v2/datasets#access) it is " All authenticated BigQuery users". Is it really "for this project"?